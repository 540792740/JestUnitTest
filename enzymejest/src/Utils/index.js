import checkPropypes from "check-prop-types"

export const findDataAttribute = (component, attr)=>{
    const wrapper = component.find(`[data-test='${attr}']`)
    return wrapper;
}

export const checkProps = (component, expectedProps) => {
    const propsErr = checkPropypes(component.propTypes, expectedProps, 'props', component.name);
    return propsErr;
};