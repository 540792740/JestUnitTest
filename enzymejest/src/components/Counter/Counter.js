import React, { Component } from 'react'

export default class Counter extends Component {
    constructor(props){
        super(props);
        this.state = {
            count : 0,
            numberOfCounter: 1
        }
    }

    makeIncreasement = (amount) =>{
        this.setState(prevState =>{
            return {
                count : prevState.count + amount
            }
        })
    }

    increment() {return this.makeIncreasement(1)}

    decrement() {return this.makeIncreasement(-1)}
    

    render() {
        return (
            <div>
                <p>Current Count: {this.state.count}</p>
                <button onClick={()=>this.decrement()} className="decrease">---</button>
                <button onClick={()=>this.increment()} className="increase">+++</button>
            </div>
        )
    }
}
