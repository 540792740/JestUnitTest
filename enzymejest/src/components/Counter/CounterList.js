import React, { Component } from 'react'
import Counter from "./Counter";

const createange = num =>{
    return Array.from(Array(num).keys());
}

export default class CounterList extends Component {
    constructor(props){
        super(props);
        this.state = {
            numberOfCounter: 2,
        }
    }
    
    buttonClick(){
        this.setState(prevState=>{
            return {numberOfCounter : prevState.numberOfCounter + 1}
        })

    }

    render() {
        const countArray = createange(this.state.numberOfCounter);
        return (
            <div>
                <p>Current Count: {this.state.count}</p>
                <button onClick={()=>this.buttonClick()} data-test="button">Add</button>
                <Counter></Counter>
                {countArray.map((value, key)=>{
                    return <Counter key={key} data-test="counter"/>
                })}
            </div>
        )
    }
}
