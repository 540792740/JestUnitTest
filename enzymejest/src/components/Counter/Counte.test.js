import React from 'react';
import { shallow } from "enzyme";
import Counter from './Counter';
import { findDataAttribute, checkProps } from "../../Utils/index"; 
import renderer from "react-test-renderer";

const setUp = (props={})=>{
 const component = shallow(<Counter {...props}/>);
 return component;
}

describe('Counter Component', () => {
    it('Match the snapshoot',()=>{
        const tree = renderer.create(<Counter/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    
    it("Start with 0 state.count",()=>{
        let component = setUp();
        const countState = component.state().count;
        expect(countState).toBe(0);
    });

    it("will increase count when click +",()=>{
        let component = setUp();
        const increaseBtn = component.find('button.increase');
        increaseBtn.simulate('click');
        const text = component.find('p').text();
        expect(text).toBe("Current Count: 1");
    });

    it("will decrease count when click +",()=>{
        let component = setUp();
        const increaseBtn = component.find('button.decrease');
        increaseBtn.simulate('click');
        const text = component.find('p').text();
        expect(text).toBe("Current Count: -1");
    });
})
