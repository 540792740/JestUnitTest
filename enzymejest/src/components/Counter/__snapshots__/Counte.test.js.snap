// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Match the snapshoot 1`] = `
<div>
  <p>
    Current Count: 
    0
  </p>
  <button
    className="decrease"
    onClick={[Function]}
  >
    ---
  </button>
  <button
    className="increase"
    onClick={[Function]}
  >
    +++
  </button>
</div>
`;
